torch/distributed/_composable/replicate.py

```diff
@@ -23,62 +25,49 @@ def replicate(
         >>> replicate(module)
     """
     torch._C._log_api_usage_once("torch.distributed.replicate")
-    _ReplicateState().mark_module(module, **kwargs)
+    _ReplicateState(ignored_modules=ignored_modules).mark_module(module, **kwargs)
     return module
 
 
-def _can_compose(module: nn.Module) -> bool:
-    r"""Check if module is composable for `replicate` API."""
-    return "fully_shard" not in _get_registry(module)
-
-
 class _ReplicateState:
-    def __init__(self) -> None:
+    def __init__(self, ignored_modules) -> None:
         self.module = None
         self.has_initialized: bool = False
         self._param_list: nn.ParameterList = nn.ParameterList()
         self.kwargs: dict = {}
+        self.ignored_modules = (
+            set(ignored_modules) if ignored_modules is not None else set()
+        )
 
     def mark_module(self, module: nn.Module, **kwargs) -> None:
-        if not _can_compose(module):
-            raise AssertionError(
-                "Cannot apply `replicate()` on a Module already managed by `fully_shard`"
-            )
         self.module = module
         replicate.state(module)._params_collected = False
         module.register_forward_pre_hook(self.forward_pre_hook)
         # TODO(@yhcharles): fix type error
         module.register_forward_hook(self.forward_post_hook)  # type: ignore[arg-type]
         self.kwargs = kwargs
 
-    def _recursive_collect_params(self, module: nn.Module) -> None:
-        # skip if managed by other APIs
-        if not _can_compose(module):
```

Comment:
when, without this, will it still work if a submodule is wrapped by `fully_shard`?